opendevin/core/config.py

```diff
@@ -277,23 +277,23 @@ def get_field_info(field):
     )
 
     # default is always present
-    default = field.default
+    default = _field.default
 
     # return a schema with the useful info for frontend
     return {'type': type_name.lower(), 'optional': optional, 'default': default}
 
 
-def load_from_env(config: AppConfig, env_or_toml_dict: dict | os._Environ):
+def load_from_env(cfg: AppConfig, env_or_toml_dict: dict | MutableMapping[str, str]):
```

Comment:
this I like to avoid confusion with regard to global config instance

Reply:
yes, avoid the conflict with the global names or imported package names.

Reply:
Well... since config is a singleton, they're all references to the one and only instance. If it seems different, maybe that's what could change.

Dunno. If you like it better, let's do it. (or even remove it as parameter directly if we can? whatever works for you)

I think these functions should become methods in that config class someday... so it'll be self actually 😅

Reply:
> or even remove it as parameter directly if we can?

it looks like a good idea.

😅 it doesn't depend on my preferences. i just want the code to be clearer for everyone.😄

Reply:
keep it as parameter, I'd say. the function can be called with made-up config instances, like in the unit tests, that have nothing to do with the singleton.

Reply:
> keep it as parameter, I'd say. the function can be called with made-up config instances, like in the unit tests, that have nothing to do with the singleton.

@tobitege that's not the case, they're all the one and only singleton. You simply can't create another instance.